hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/common/transport/server/GrpcXceiverService.java

```diff
@@ -121,6 +133,9 @@ public void onNext(ContainerCommandRequestProto request) {
           if (popStream != null) {
             IOUtils.close(LOG, popStream);
           }
+          if (context != null) {
```

Comment:
Similar context creation/close should be done for  `ContainerStateMachine#readStateMachineData`?

Reply:
@duongkame , Good catch!  We should if we make DispatcherContext closable for releasing buffers.

On a second thought, HDDS-7117 already has changed using `MappedByteBuffer` instead of allocating buffers when the size is small than `ozone.chunk.read.mapped.buffer.threshold` (default 32KB).  So, it seems that we don't have do anything for read.  Could you test HDDS-9536 with the current code?

Reply:
That sounds nice. @szetszwo , will update.